atlas/src/main/java/org/georchestra/atlas/converters/AtlasJobConverter.java

```diff
@@ -0,0 +1,61 @@
+package org.georchestra.atlas.converters;
+
+import org.apache.camel.Converter;
+import org.apache.camel.Exchange;
+
+import org.georchestra.atlas.AtlasJob;
+import org.json.JSONException;
+
+import java.io.*;
+
+@Converter
+@SuppressWarnings("unused")
+public class AtlasJobConverter {
+
+    private static String toString(InputStream property) throws IOException {
+
+        property.reset();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(property));
+
+        StringBuilder rawString = new StringBuilder();
+        String line;
+        while((line = reader.readLine()) != null)
+            rawString.append(line);
+        return rawString.toString();
+    }
+
+    @Converter
+    public AtlasJob toAtlasJob(InputStream is, Exchange exchange) throws IOException, JSONException {
+
+        String body = AtlasJobConverter.toString(is);
+        return this.toAtlasJob(body, exchange);
+    }
+
+    @Converter
+    public AtlasJob toAtlasJob(String body, Exchange exchange) throws IOException, JSONException {
+
+        if(exchange.getProperties().containsKey("jobId")) {
+//            Long jobId = exchange.getProperty("rawJson", Long.class);
+            AtlasJob res = new AtlasJob(body);
+//            res.setId(jobId);
+            return res;
+        } else {
+            return new AtlasJob(body);
+        }
```

Comment:
I cannot remember the logic beyond this method. How about simplifying it, since the if / else seems to do the exact same thing ?

```java
@Converter
public AtlasJob toAtlasJob(String body, Exchange exchange) throws IOException, JSONException {
   return new AtlasJob(body);
}
```